api/analyzedphenotypes.ontology.api.inc

```diff
@@ -0,0 +1,391 @@
+  if (is_null($property)) {
+    // Default.
+    $property = 'default';
+  }
+
+  // Process request.
+  $command = array(
+    'map_ontology_trait' => array(
+      'func'  => 'ap_set_ontology',
+      'param' => array('trait_cvterm' => $parameter['trait_cvtermid'], 'ontology_cvterm_id' => $parameter['ontology_cvtermid']),
+      'opts'  => null,
+    ),
+    'match_ontology' => array(
+      'func'  => 'ap_match_ontology',
+      'param' => $parameter['ontology'],
+      'opts'  => $parameter['genus'],
+    ),
+    'suggest' => array(
+      'func'  => 'ap_suggest_ontology',
+      'param' => array('trait' => $parameter['trait_name'], 'genus' => $parameter['genus']),
+      'opts'  => null,
+    ),
+  );
+
+  // Execute function.
+  call_user_func($command[$property]['func'], $command[$property]['param'], $command[$property]['opts']);
+}
+
+// * @TODO: leave out when api reorganize/update is complete and use functions below.
+function analyzedphenotypes_defaultontologydefine() {
+  return ap_define_defaultontology();
+}
+
+// * @TODO: leave out when api reorganize/update is complete and use functions below.
+function analyzedphenotypes_defaultontologyload() {
+  return ap_load_defaultontology();
+}
+
+
+///////
+
+
+/**
+ * FUNCTION:
+ * Insert into chado.cvterm or test existence of a cvterm.
+ * and return cvterm id to be set as default value in configuration page.
+ * Dependencies: ap_define_defaultontology().
+ *
+ * @return
+ *   An array, controlled vocabulary terms cvterm id.
+ *
+ * @see hook_install() implementation in install.inc.
+ */
+function ap_load_defaultontology() {
+  $ontology = ap_define_defaultontology();
+
+  // This array to hold default controlled vocabulary terms used in this system.
+  $ontology_load = array();
+
+  // Each controlled vocabulary - cv.
+  foreach($ontology['terms'] as $abbrev => $details) {
+    // Each term in a Controlled Vocabulary cv.
+    foreach($details['terms'] as $term) {
+      // Fetch the term, see if previously added.
+      $a = array(
+        'name' => $term['name'],
+        'cv_id' => array('name' => $details['name']),
+      );
+
+      if (function_exists('chado_get_cvterm')) {
+        $t = chado_get_cvterm($a);
+      }
+      else {
+        $t = tripal_get_cvterm($a);
+      }
+
+      if (!$t) {
+        // Term not in chado.cvterm. Insert!
+        // #1. Controlled vocabulary.
+        $ins_cv = tripal_insert_cv($details['name'], $details['definition']);
+
+        // #2. Terms.
+        $term['cv_name'] = $details['name'];
+        $ins_cvterm = tripal_insert_cvterm($term);
+      }
+
+      // CV TERM NAME.
+      $name = ($t) ? $t->name : $ins_cvterm->name;
+      // Match predefined variable to cvterm saved.
+      $map_term = array_search($name, $ontology['map_term']);
+
+      // Save term to be used as configuration page default settings.
+      // eg. ap_location = 1231.
+      $ontology_load[$map_term] = ($t) ? $t->cvterm_id : $ins_cvterm->cvterm_id;
```

Comment:
This assumes you were able to insert the cvterm properly. You should have an 
```php
if (!$ins_cvterm) {
  tripal_report_error(...);
}
```
above for easier debugging if something goes wrong :-)